### PR TITLE
feat(string): add string replace

### PR DESCRIPTION
**Pull request by fengmk2**

### CODE REVIEW

File: lib/string.js [hunk 1]
```diff
@@ -58,3 +58,11 @@ exports.splitAlwaysOptimized = function splitAlwaysOptimized() {
   }
   return needs;
 };
+
+exports.replace = function replace(str, substr, replacement) {
+  var index = str.indexOf(substr);
```
Comment (dead-horse) [inline lib/string.js]:
```
var replaceFunction = typeof replacement === 'function' 
  ? replacement
  : function () {
    return replacement;
  };

  return str.replace(substr, replaceFunction);
}
```

这样应该可以保证正则也支持吧

**Comment by fengmk2** [comment]
now support all params like `String.prototype.replace`

**Comment by dead-horse** [comment]
+1

**Comment by popomore** [comment]
Isn't it a feature of string?

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/replace#Specifying_a_string_as_a_parameter

**Comment by dead-horse** [comment]
but in some situation we don't want to use this "feature"

**Pull request merged**